### PR TITLE
[14.0][PERF] product: template name search with no active variant

### PR DESCRIPTION
**Pull request by mt-software-de**
We are facing a performance issue. If we are searching for mrp.bom of a specifict product search query will never finish. 
This is because of the not performant search of templates with no active products. 
https://github.com/odoo/odoo/blob/4c8d544c8c50bbc1ff646edcc4b130c0ccf0ef91/addons/product/models/product_template.py#L502

In our database there are 620236 products and 579345 templates. The search https://github.com/odoo/odoo/blob/4c8d544c8c50bbc1ff646edcc4b130c0ccf0ef91/addons/product/models/product_template.py#L506 will never finish.

Therefor i switch to an normal sql query. 

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by BHanq** [comment]
Hello, 

Thanks for your PR, you are right that this feature need an improvement. 

We also refactored this method (https://github.com/odoo/odoo/pull/143543) which should be merge asap. With the same conclusion (while using a not exist instead of left join+in). But we prefer to stick with the ORM. A raw sql query doesn't take into account the ir.rules and doesn't flush the data (https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#sql-execution).

File: addons/product/models/product_template.py [hunk 1]
```diff
@@ -470,6 +470,44 @@ def name_get(self):
...
+                product_product as p
+            ON
+                p.product_tmpl_id = pt.id
+            GROUP BY
+                pt.id
+        ) as sub
+        WHERE
+            false = all(sub.active);"""
+        self.env.cr.execute(query)
+        ids = set([r[0] for r in self.env.cr.fetchall()])
```
[the remark this answers is not in the document]
Comment (mt-software-de) [reply addons/product/models/product_template.py:494]: Yes of course. With the first query i am getting all product_templates which no product.product.
And this query returns the product.template ids which have only inactive product.product. 
But i didn't liked it at all to use to queries, your approach is much more convenient thx.

**Comment by mt-software-de** [comment]
> Hello,
> 
> Thanks for your PR, you are right that this feature need an improvement.
> 
> We also refactored this method (#143543) which should be merge asap. With the same conclusion (while using a not exist instead of left join+in). But we prefer to stick with the ORM. A raw sql query doesn't take into account the ir.rules and doesn't flush the data (https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#sql-execution).

I cherry-picked your commit.

**Comment by pivi-odoo** [comment]
Superseded by https://github.com/odoo/odoo/pull/143543 
Odoo 14 is now EOL, so DBs needs to migrate to a newer version and will get the patch.
Older on-premise DBs can cherry-pick the commit if they really need to.

**Pull request closed without merging**